module/scripts/combat/effects.js

```diff
@@ -0,0 +1,136 @@
+
+async function createDamageMessage(actor, damageAmount, content, remedy = '', damage) {
+  // Create a message containing the damage data
+  const messageData = {
+    content: `<div class="damage-message">${content}: ${damageAmount}<br>${remedy}</div>`,
+    speaker: ChatMessage.getSpeaker({ actor: actor }),
+    flags: {
+      'TheWitcherTRPG': {
+        damage: damage,
+      },
+    },
+  };
+
+  // Create the chat message
+  const message = await ChatMessage.create(messageData);
+
+  // Return the message ID
+  return message.id;
+}
+
+async function applyFireDamageToEquippedItems(actor, damage) {
+  // Update stopping power for equipped armor
+  for (const item of actor.items.filter(i => i.type === 'armor' && i.system.equipped)) {
+    const locationsCovered = ['head', 'torso', 'leftArm', 'rightArm', 'leftLeg', 'rightLeg'];
+    for (const location of locationsCovered) {
+      const stoppingField = `${location}Stopping`;
+      if (item.system[stoppingField] !== undefined) {
+        const stoppingPower = item.system[stoppingField] ?? 0;
+        if (stoppingPower > 0) {
+          const newStoppingPower = Math.max(stoppingPower - damage, 0);
+          await item.update({ [`system.${stoppingField}`]: newStoppingPower });
+          ui.notifications.info(`${item.name} loses ${damage} stopping power in ${location}!`, { permanent: false, delay: 500 });
+        }
+      }
+    }
+  }
+
+  // Update reliability for equipped weapons
+  for (const item of actor.items.filter(i => i.type === 'weapon' && i.system.equipped)) {
+    const reliabilityValue = item.system?.reliable ?? 0;
+    const newReliability = Math.max(reliabilityValue - 1, 0);
+    await item.update({ "system.reliable": newReliability });
+    ui.notifications.info(`${item.name} loses 1 reliability!`, { permanent: false, delay: 1000 });
+  }
+}
+
+async function applyFire(actor) {
+  const fireDamage = -5;
+  const content = 'Fire Damage';
+  const remedy = 'Damages all equipped items. A full round action will remove the effect.';
+
+  // Declare and initialize the damage object
+  const damage = {
+    location: 'torso',
+    damageProperties: {
+      damageToAllLocations: true,
+    },
+  };
+
+  // Create the damage message and get the message ID
+  const messageId = await createDamageMessage(actor, fireDamage, content, remedy, damage);
+
+  // Apply the fire damage using the message ID
+  ApplyNormalDamage(actor, fireDamage, messageId, { skipDialog: true });
+
+  // Ensure effects are applied to the actor
+  await applyFireDamageToEquippedItems(actor, 1);
+}
+
+async function applyPoison(actor) {
+  const poisonDamage = -3;
+  const content = 'Poison Damage';
+  const remedy = 'You must pass a DC:15 Endurance check that takes 1 action.';
+
+  // Declare and initialize the damage object
+  const damage = {
+    location: 'torso',
+    damageProperties: {
+      bypassesNaturalArmor: true,
+      bypassesWornArmor: true,
```

Comment:
I noticed that these properties are leading to a generation of shield. Unsure if this is a bug or if i misunderstood these.

Will look into that a bit later but maybe someone knows already.

Reply:
Shields should only be generated via https://github.com/witchertrpg-foundryvtt/TheWitcherTRPG/blob/main/module/scripts/chat.js#L43